src/test/fixtures/sources.ts

```diff
@@ -22,35 +28,37 @@ export const mockSourcesByProvider = (provider) => {
         source_type_id: '3',
         uid: '73d5694c-7a28-417e-9fca-55840084f508',
       },
-    ];
-  }
+    ],
+  };
+  return mockSources[provider as keyof typeof mockSources];
 };
 
-export const mockUploadInfo = (sourceId) => {
-  if (sourceId === '666') {
-    return {
+export const mockUploadInfo = (
+  sourceId: string
+): GetSourceUploadInfoApiResponse => {
+  const mockInfo = {
+    '666': {
       provider: 'azure',
       azure: {
         tenant_id: '2fd7c95c-0d63-4e81-b914-3fbd5288daf7',
         subscription_id: 'dfb83267-e016-4429-ae6e-b0768bf36d65',
         resource_groups: ['myResourceGroup1', 'testResourceGroup'],
       },
-    };
-  } else if (sourceId === '667') {
-    return {
+    },
+    '667': {
       provider: 'azure',
       azure: {
         tenant_id: '73d5694c-7a28-417e-9fca-55840084f508',
         subscription_id: 'a66682d2-ce3e-46f7-a127-1d106c34e10c',
         resource_groups: ['theirGroup2'],
       },
-    };
-  } else if (sourceId === '123') {
-    return {
+    },
+    '123': {
       provider: 'aws',
       aws: {
         account_id: '123456789012',
       },
-    };
-  }
+    },
+  };
+  return mockInfo[sourceId as keyof typeof mockInfo];
```

Comment:
Similar to the previous PR, I think it would be more consistent to type the mockInfo object and remove `as keyof typeof mockInfo`.